build.gradle

```diff
@@ -1,36 +1,216 @@
 
-    maven {
-        url "https://mvnrepository.com/artifact/org.apache.hive/hive-metastore"
-    }
 
-    maven {
-        url 'https://mvnrepository.com/artifact/org.apache.kafka/kafka-clients'
-    }
+//    exclude group:'org.apache.logging.log4j', module: 'log4j'
+//    exclude group: 'org.apache.logging.log4j', module: 'log4j-slf4j-impl'
+}
+
+group = 'org.odpi.egeria'
+version = '1.0-SNAPSHOT'
+ext.name = 'New Egeria code'
+description = 'New code for Egeria'
+
+ext {
+    egeriaversion = '3.8'
+    slf4jVersion = '1.7.36'
+    jacksonVersion = '2.13.2'
+    jupiterVersion = '5.8.2'
+    httpclientVersion = '4.5.13'
+    springwebVersion = '5.3.19'
+    hiveversion = '3.1.3'
 }
 
+// For a single module, add here. For multiple modules, refactor and use constraints (see main egeria build.gradle for an example)
 dependencies {
-    testImplementation 'org.junit.jupiter:junit-jupiter-api:5.8.2'
-    testRuntimeOnly 'org.junit.jupiter:junit-jupiter-engine:5.8.2'
 
-    implementation "org.odpi.egeria:repository-services-apis:3.4"
-    implementation "org.odpi.egeria:audit-log-framework:3.4"
-    implementation "org.odpi.egeria:open-connector-framework:3.4"
-    implementation "org.odpi.egeria:connector-configuration-factory:3.4"
+    // Only used for build - already present in server chassis at runtime - similar to 'provided' with maven
+    compileOnly "org.odpi.egeria:audit-log-framework:${egeriaversion}"
 
+    // Needed for build and run
+    implementation "org.slf4j:slf4j-api:${slf4jVersion}"
 
 
+    implementation "com.fasterxml.jackson.core:jackson-databind:${jacksonVersion}"
+    implementation "com.fasterxml.jackson.core:jackson-annotations:${jacksonVersion}"
+    implementation "com.fasterxml.jackson.core:jackson-core:${jacksonVersion}"
+    implementation "org.springframework:spring-core:${springwebVersion}"
+    implementation "org.junit.jupiter:junit-jupiter:${jupiterVersion}"
+    implementation "org.odpi.egeria:open-connector-framework:${egeriaversion}"
+    implementation "org.odpi.egeria:open-metadata-types:${egeriaversion}"
+    implementation "org.odpi.egeria:audit-log-framework:${egeriaversion}"
+    implementation "org.odpi.egeria:repository-services:${egeriaversion}"
+    implementation "org.odpi.egeria:repository-services-implementation:${egeriaversion}"
+    implementation "org.odpi.egeria:admin-services-api:${egeriaversion}"
+    implementation "org.odpi.egeria:connector-configuration-factory:${egeriaversion}"
+    implementation "org.odpi.egeria:repository-services-apis:${egeriaversion}"
+    implementation "org.odpi.egeria:admin-services-api:${egeriaversion}"
+    implementation "org.odpi.egeria:audit-log-framework:${egeriaversion}"
+    // Hive
+//    implementation 'org.apache.thrift:libthrift:0.9.3'
+//    implementation 'org.apache.thrift:libfb303:0.9.3'
+//    implementation "org.apache.hive:hive-common:3.1.2"
+    implementation "org.apache.hadoop:hadoop-common:3.3.3"
```

Comment:
The same issue we've seen with Egeria - OSSIndex (sonatype) showing this version has clean, but lift isn't. Apparently they have two data sources and are trying to rationalise....

Suggest doing what you can for now, but we should probably take this up as an issue in its own right to see what we can do to document/mitigate.
<img width="1397" alt="Screenshot 2022-07-22 at 12 45 12" src="https://user-images.githubusercontent.com/7292002/180432876-3eeebedb-b04a-40d1-b87c-7cdcfaff0499.png">

Applies to other similar observations

Reply:
I've posted the question to https://community.sonatype.com/t/ossindex-shows-component-clean-but-when-used-as-a-dependency-lift-scanner-says-severe-vulnerability/8696/9?u=nigel.l.jones

Reply:
@planetf1 It appears clear on OSSI because OSSI is only telling you about vulnerabilities that Hadoop-common directly has reported against it rather than vulnerabilities of the dependencies that Hadoop-common brings in. 

Lift performs a build and can therefore work out the entire dependency tree and is saying that Hadoop-common 3.3.3 is bringing in log4j 1.2.17 for example which is know to have critical vulnerabilities.

Lift generates a BOM that you can use to work out where dependencies and vulnerabilities are coming from : https://lift.sonatype.com/results/github.com/odpi/egeria-connector-hivemetastore/01G8JSPTY6X5S8H2C15DWGCFJZ?tab=dependencies

<img width="1025" alt="Screen Shot 2022-07-22 at 10 02 49 AM" src="https://user-images.githubusercontent.com/13700083/180444761-446c9d20-67c4-4c92-8b9c-dac30712efed.png">

In this case it looks like Lift has correctly found that Hadoop is bringing in older version of zookeeper which in turn relies on a vulnerable version of log4j

Reply:
@jlstephens89 thanks for the explanation - we do still have nexus IQ setup for other projects - running off merge builds - we're just looking at other tooling & the PR injection with lift is very useful to catch problems earlier.

Hadoop is a particular annoyance as it does bring in a lot, and is a little stale. But we can look at mitigations such as  forcing transitive version resolution (it not already done) or excluding if not needed!

Reply:
@planetf1 shouldnt  exclude group: 'log4j', module: 'log4j:1.2.17' solve this issue?

Reply:
I am hoping to merge this change.

Reply:
I would think so, BUT we need to check. And if so one wonders why the scanner is still reporting it.

What does `gradlew dependencies -q` show. Is that library definately excluded?

Then, does it appear within any of the internal jars (recursive unpack to inspect this)?

That would be the way to determine if log4J is still present.

Since this code is still being developed maybe it's reasonable to merge anyway, but to open up an issue to track any outstanding issues - such as if the answer to either of the two above is yes, or the check isn't done?

Reply:
@planetf1, it was still in the dependancies - I have now excluded it - log4j is no longer in this connectors dependancies according to `gradlew dependencies -q` , so the only log4j that could be picked up is that from Egeria.